### ISSUE TITLE
WCAG 1.3.1 test 2 too strict?

### ISSUE THREAD

**Issue by jzuidweg** [veraPDF/veraPDF-wcag-algs#225]
[Test document 2](https://github.com/veraPDF/veraPDF-wcag-algs/files/9111817/test-document-2.pdf) fails WCAG 1.3.1 test 2 (_Space characters are used for formatting purpose_). This is technically correct. There are no false positives: the test correctly identifies places where spaces have no function and line breaks are used to create vertical space.

However, you dould argue that it's a bit over the top to generate an error for a single or double space directly after a full stop. It's true that this is bad practice, but it generates a lot of errors which are not considered very important by people who use screen readers (which usually ignore spaces after a full stop).

My proposal for implementing this test would be:

- Ignore single or double spaces directly after a full stop. Do not generate an error in this case.
- Generate errors for 3 or more consecutive spaces.
- Generate errors for repeated line breaks following empty lines, or lines with only spaces.

**Comment by bdoubrov** [comment]
We already generate errors only if there are 3 or more repeated spaces. This would cover the first two bullet points.

Line breaks and empty lines normally do not normally appear in PDF content or structure. What can be detected additionally:
- paragraphs (in fact any structure elements) that contain only spaces (even 1 or 2)
- abnormally large vertical spacing between text lines

**Comment by jzuidweg** [comment]
Thanks for the answer, this sounds good. My question is: why does the algorithm still seem to point to single spaces after a line end with full stop ('. ')? This appears to occur in test document 2.

**Comment by irinamavrina** [comment]
@jzuidweg Hi, let me answer your question)
The single spaces you're referring to, are not actually single (they look like big single spaces, but if you look at another case with repeating spaces you'll see, that actual single spaces are narrower than spaces in question), so really there are 3 successing spaces and that's why algorithm points to them.
